realm/realm-library/src/objectServer/java/io/realm/SyncUserInfo.java

```diff
@@ -24,50 +27,41 @@
  */
 
 public class SyncUserInfo {
-    private final String provider;
-    private final String providerUserIdentity;
     private final String identity;
     private final boolean isAdmin;
+    private final Map<String, String> metadata;
 
-    private SyncUserInfo(String provider, String providerUserIdentity, String identity, boolean isAdmin) {
-        this.provider = provider;
-        this.providerUserIdentity = providerUserIdentity;
+    private SyncUserInfo(String identity, boolean isAdmin, Map<String, String> metadata) {
         this.identity = identity;
         this.isAdmin = isAdmin;
+        this.metadata = Collections.unmodifiableMap(metadata);
     }
 
     static SyncUserInfo fromLookupUserIdResponse(LookupUserIdResponse response) {
-        return new SyncUserInfo(response.getProvider(), response.getProviderId(), response.getUserId(), response.isAdmin());
+        return new SyncUserInfo(response.getUserId(), response.isAdmin(), response.getMetadata());
     }
 
     /**
-     * @return identity providers {@link io.realm.SyncCredentials.IdentityProvider} which manages the user represented by this user info instance.
-     */
-    public String getProvider() {
-        return provider;
-    }
-
-    /**
-     * @return The username or identity issued to this user by the authentication provider.
-     */
-    public String getProviderUserIdentity() {
-        return providerUserIdentity;
-    }
-
-    /**
-     * @return The identity issued to this user by the Realm Object Server.
+     * @return the identity issued to this user by the Realm Object Server.
      */
     public String getIdentity() {
         return identity;
     }
 
     /**
-     * @return Whether the user is flagged on the Realm Object Server as an administrator.
+     * @return whether the user is flagged on the Realm Object Server as an administrator.
      */
     public boolean isAdmin() {
         return isAdmin;
     }
 
+    /**
+     * @return the metadata associated with this user.
```

Comment:
maybe a example about what kind of map could be returned? i am bit unclear when read this doc.

Reply:
It is just a `key/value` map. I'll expand the description